setup.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+
+from glob import glob
+import os
+import sys
+
+from setuptools import setup, Extension
+from Cython.Build  import cythonize
+
+if sys.argv[-1] == 'setup.py':
+    print('To install, run \'python setup.py install\'')
+
+if sys.version_info[:2] < (2, 7):
+    print(
+        'networkx-lemon requires Python version 2.7 or later' +
+        ' ({}.{} detected).'.format(*sys.version_info[:2]))
+    sys.exit(-1)
+
+libraries = [
+    ('lemon', {'sources': glob('src/lemon/*.cc') +
+                          glob('src/lemon/bits/*.cc'),
```

Comment:
`Random` is used in only `grosso_locatelli_pullan_mc.h`. So we do not need it immediately either. As for `base.cc`, it defines the `Tolerance` constants used in various flow algorithms, which may be some short-term targets. I would suggest that we compile just that one file into `liblemon`.